examples/tv-app/linux/AppImpl.cpp

```diff
@@ -132,17 +132,94 @@ DECLARE_DYNAMIC_ATTRIBUTE(ZCL_CHANNEL_LIST_ATTRIBUTE_ID, ARRAY, kDescriptorAttri
     DECLARE_DYNAMIC_ATTRIBUTE(ZCL_CHANNEL_CURRENT_CHANNEL_ATTRIBUTE_ID, STRUCT, 1, 0),             /* current channel */
     DECLARE_DYNAMIC_ATTRIBUTE_LIST_END();
 
+constexpr CommandId keypadInputIncomingCommands[] = {
```

Comment:
This is unfortunate that we've ended up in this place, where the application author has to carefully create this list to match what the cluster impl expects.

The cluster impl **is** the authority on knowing which commands it should/should not expect. If there are feature flags that alter that behavior, those should be protected with compile/dynamic feature flags (if any).

Anyways...

Reply:
The cluster impl might be able to handle commands A, B, C, but the application only wants to enable commands A, B on endpoint 1 and commands B, C on endpoint 2.  The application is the only thing that knows which commands are actually enabled on a given endpoint.

In ZAP, this is accomplished by selecting things in the ZAP UI.  When using external endpoints not defined via ZAP, it has to be defined somehow, by the application.